staging/src/k8s.io/apimachinery/pkg/apis/meta/fuzzer/fuzzer.go

```diff
@@ -89,6 +90,21 @@ func genericFuzzerFuncs(codecs runtimeserializer.CodecFactory) []interface{} {
 				panic(fmt.Sprintf("Failed to encode object: %v", err))
 			}
 
+			if json.Valid(bytes) {
```

Comment:
This is sort of masking a problem by normalizing in the fuzzer something that would actually break anyone populating RawExtension from a typed JSON encoding in real life, right? I think this is the issue noted in https://github.com/kubernetes/kubernetes/issues/73655

It looks like we have a customer fuzzer for ControllerRevision as well that forces ordered json and masks the issue:
```go
		func(r *apps.ControllerRevision, c fuzz.Continue) {
			c.FuzzNoCustom(r)
			// match the fuzzer default content for runtime.Object
			r.Data = runtime.RawExtension{Raw: []byte(`{"apiVersion":"unknown.group/unknown","kind":"Something","someKey":"someValue"}`)}
		},
```


This probably should have a TODO to drop this test-only compensation and reference the issue to actually resolve. Do we know if CBOR will make this worse by putting non-JSON data in RawExtension fields that will *require* JSON normalization that will shuffle the order of fields?

Reply:
Another example is https://github.com/kubernetes/kubernetes/blob/08e6500cd92c367fb641fd421a592ca920634122/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/construct.go#L35-L39. I agree it deserves mentioning here too.

Reply:
rather than fuzzing RawExtension differently for *every* consumer, let's start by applying normalization to the containing types that use RawExtension and use the round-trip failures as an opportunity to identify those types and see if there's other validation / fixup / backstop work we need to do to make those containing types safe (e.g. https://github.com/kubernetes/kubernetes/commit/d6d78c5581537567c742c50188f006422f4b6e98)

I would also suggest making a helper `NormalizeJSONRawExtension` helper function that those can all call to do this normalization